packages/csharp/src/components/interface/method.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+  Block,
+  Children,
+  MemberDeclaration,
+  refkey,
+  Refkey,
+  Scope,
+} from "@alloy-js/core";
+import {
+  AccessModifiers,
+  computeModifiersPrefix,
+  getAccessModifier,
+  makeModifiers,
+} from "../../modifiers.js";
+import { useCSharpNamePolicy } from "../../name-policy.js";
+import { CSharpOutputSymbol } from "../../symbols/csharp-output-symbol.js";
+import { CSharpMemberScope, useCSharpScope } from "../../symbols/scopes.js";
+import { ParameterProps, Parameters } from "../Parameters.jsx";
+
+/** Method modifiers. Can only be one. */
+export interface InterfaceMethodModifiers {
+  readonly new?: boolean;
+}
+
+export const getMethodModifier = makeModifiers<InterfaceMethodModifiers>([
+  "new",
+]);
+
+// properties for creating a method
+export interface InterfaceMethodProps
+  extends AccessModifiers,
+    InterfaceMethodModifiers {
+  name: string;
+  refkey?: Refkey;
+  children?: Children;
+  parameters?: Array<ParameterProps>;
+  returns?: Children;
+}
+
+// a C# interface method
+export function InterfaceMethod(props: InterfaceMethodProps) {
+  const name = useCSharpNamePolicy().getName(props.name, "class-method");
+  const scope = useCSharpScope();
+  if (scope.kind !== "member" || scope.name !== "interface-decl") {
+    throw new Error(
+      "can't define an interface method outside of an interface scope",
+    );
+  }
+
+  const methodSymbol = new CSharpOutputSymbol(name, {
+    scope,
+    refkeys: props.refkey ?? refkey(props.name),
+  });
+
+  // scope for method declaration
+  const methodScope = new CSharpMemberScope("method-decl", {
+    owner: methodSymbol,
+  });
+
+  const params =
+    props.parameters ? <Parameters parameters={props.parameters} /> : "";
+
+  const modifiers = computeModifiersPrefix([
+    getAccessModifier(props),
+    getMethodModifier(props),
+  ]);
+  // note that scope wraps the method decl so that the params get the correct scope
+  return (
+    <MemberDeclaration symbol={methodSymbol}>
```

Comment:
should we not have a csharp `MemeberDeclaration` type that just take the name instead?